python-checks/src/main/java/org/sonar/python/checks/cdk/AbstractCdkResourceCheck.java

```diff
@@ -171,4 +176,42 @@ protected static Predicate<Expression> isFqn(String fqnValue) {
       .isPresent();
   }
 
+  protected static Predicate<Expression> isString(String str) {
+    return expression ->  Optional.of(expression)
+      .filter(expr -> expr.is(Tree.Kind.STRING_LITERAL)).map(StringLiteral.class::cast)
+      .filter(stringLiteral -> stringLiteral.trimmedQuotesValue().equals(str))
+      .isPresent();
+  }
+
+  protected static Predicate<Expression> isDictionaryWithKeyValue(SubscriptionContext ctx, String key, String val) {
+    return expression -> expression.is(Tree.Kind.DICTIONARY_LITERAL)
+      && asDictionaryKeyValuePairs(expression)
+      .filter(isKey(key))
```

Comment:
Keys in dictionaries are also expressions. So they could also be a referenced value. It's an edge case, but can be handled similarly to the resolution of the values.